05-JWT-Basics/starter/middleware/auth.js

```diff
@@ -0,0 +1,23 @@
+const jwt = require("jsonwebtoken");
+const { UnauthenticatedError } = require("../errors");
+
+const authenticationMiddleware = async (req, res, next) => {
+  const authHeader = req.headers.authorization;
+
+  if (!authHeader || !authHeader.startsWith("Bearer ")) {
+    throw new UnauthenticatedError("No token provided");
+  }
+
+  const token = authHeader.split(" ")[1];
```

Comment:
Extra challenge: can we use regular expressions to parse the tokens instead? Again, slack me anytime for hints!!